libs/core/src/schemas/events.schemas.ts

```diff
@@ -46,3 +46,52 @@ export const DiscordMessageCreated = z.object({
     z.literal('comment-create'),
   ]),
 });
+
+// All events should carry this common metadata
+export const EventMetadata = z.object({
+  createdAt: z.date().describe('When the event was emitted'),
+  // TODO: TBD
+  // aggregateType: z.enum(Aggregates).describe("Event emitter aggregate type")
+  // aggregateId: z.string().describe("Event emitter aggregate id")
+  // correlation: z.string().describe("Event correlation key")
+  // causation: z.object({}).describe("Event causation")
+});
+
+// on-chain contest manager events
+export const ContestManagerDeployed = EventMetadata.extend({
+  namespace: z.string().describe('Community namespace'),
+  contest: z.string().describe('Contest manager address'),
+  interval: z.number().int().positive().describe('Recurring constest interval'),
+  oneOff: z.boolean().describe('Flags one-off contests'),
+}).describe('When a contest contract manager gets deployed');
+
+export const ContestStarted = EventMetadata.extend({
+  contest: z.string().describe('Contest manager address'),
+  contestId: z.number().int().describe('New contest instance id'),
+  startTime: z.date().describe('Contest start time'),
+  endTime: z.date().describe('Contest end time'),
+}).describe('When a contest instance gets started');
+
+export const ContestContentAdded = EventMetadata.extend({
+  contest: z.string().describe('Contest manager address'),
+  contestId: z.number().describe('Contest instance id'),
+  contentId: z.number().describe('New content id'),
+  creator: z.string().describe('Address of content creator'),
+  url: z.string().describe('Content url'),
+}).describe('When new content is added to a running contest');
+
+export const ContestContentUpvoted = EventMetadata.extend({
```

Comment:
(To be discussed - not yet determined): Inside Protocol, I would like discuss recurring events can be turn into backend listener for each block.

@Rotorsoft proposed that we can intitiate a bool flag to trigger backend each block to start parsing process if there is a new content or contest voting happened in the company.